mapsforge-map-android/src/main/java/org/mapsforge/map/android/graphics/AndroidPaint.java

```diff
@@ -146,11 +151,118 @@ public int getTextHeight(String text) {
+    }
+
     @Override
     public int getTextWidth(String text) {
         return (int) this.paint.measureText(text);
     }
 
+    @Override
+    public int getTextWidth(String text, boolean includePadding) {
+        int retVal = getTextWidth(text);
+
+        if (includePadding && retVal > 0) {
+            retVal += getFontPadding();
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Uses StaticLayout to measure text accurately, and respects Spans.
+     * Returns width of the widest line.
+     */
+    @SuppressWarnings("deprecation")
+    @Override
+    public int getTextWidth(final String text, final int widthMax, boolean includePadding) {
+        int retVal = 0;
+
+        if (text != null && widthMax >= 0) {
+
+            final TextPaint myPaint;
+            if (this.paint instanceof TextPaint) {
+                myPaint = (TextPaint) this.paint;
+            } else {
+                myPaint = new TextPaint(this.paint);
+            }
+
+            final StaticLayout layout = new StaticLayout(text, myPaint, widthMax, Layout.Alignment.ALIGN_NORMAL, 1, 0, includePadding);
+
+            final int lineCount = layout.getLineCount();
+            float maxWidth = 0;
+
+            for (int i = 0; i < lineCount; i++) {
+                final float lw = layout.getLineWidth(i);
+
+                if (maxWidth < lw) {
+                    maxWidth = lw;
+                }
+            }
+
+            retVal = (int) maxWidth;
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Uses StaticLayout to measure text accurately, and respects Spans.
+     * Returns width of the widest line.
+     */
+    public static int getTextWidth(final StaticLayout layout, boolean includePadding) {
+        int retVal = 0;
+
+        if (layout != null) {
+
+            final int lineCount = layout.getLineCount();
+            double maxWidth = 0;
+
+            for (int i = 0; i < lineCount; i++) {
+                final float lw = layout.getLineWidth(i);
+
+                if (maxWidth < lw) {
+                    maxWidth = lw;
+                }
+            }
+
+            retVal = (int) maxWidth;
+
+            if (includePadding && retVal > 0) {
+                retVal += getFontPadding(layout.getPaint());
+            }
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Returns total vertical font padding (top+bottom). The value can also be used for horizontal padding.
+     */
+    @Override
+    public int getFontPadding() {
+        return getFontPadding(this.paint);
+    }
+
+    /**
+     * Returns total vertical font padding (top+bottom). The value can also be used for horizontal padding.
+     * The value is computed as (ascent - top + bottom).
+     */
+    public static int getFontPadding(android.graphics.Paint paint) {
+        final android.graphics.Paint.FontMetricsInt fontMetricsInt = paint.getFontMetricsInt();
+
+        return fontMetricsInt.ascent - fontMetricsInt.top + fontMetricsInt.bottom;
```

Comment:
@Sublimis Please explain the padding calculation details.

So we can apply a similar calculation in the `AwtPaint`.